src/discriminant-calculator.c

```diff
@@ -0,0 +1,70 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <math.h>
+
+struct CalculationResult {
+	int result;
+	int isSuccess;
+	int x1;
+	int x2;
+	char message[100];
+};
+
+struct CalculationResult rootCalculator(int discriminant, int a, int b, int c) {
+        struct CalculationResult result;
+	
+	if (discriminant >= 0) {
+		result.x1 = (-b + sqrt(discriminant) / (2 * a));
+		result.x2 = (-b - sqrt(discriminant) / (2 * a));
+		result.isSuccess = 1;
+		snprintf(result.message, sizeof(result.message), "Real roots found");
+	} else {
+		result.isSuccess = 0;
+		snprintf(result.message, sizeof(result.message), "Complex roots found");
+	}
+
+	return result;
+}
+
+struct CalculationResult discriminantCalculator(int a, int b, int c) {
+	struct CalculationResult result;
+
+	int res = pow(b, 2) - 4 * a * c;
+	printf("Discriminant = %d \n", res);
+
+	if (res >= 0) {
+		result = rootCalculator(res, a, b, c);
+	} else {
+		result.isSuccess = 0;
+		snprintf(result.message, sizeof(result.message), "Discriminant < 0, complex roots");
+	}
+
+	return result;
+}
+
+void validateInput(int argc, char* argv[]) {
```

Comment:
Why this function do the discriminantCalculator?
It is not a responsibility of this function

Reply:
done